tests/common/devices.py

```diff
@@ -1939,6 +1982,167 @@ def is_service_running(self, service_name, docker_name):
             )
         return self.sonichost.is_service_running(service_name, docker_name)
 
+    def ping_v4(self, ipv4, count=1):
+        """
+        Returns 'True' if ping to IP address works, else 'False'
+        Args:
+            IPv4 address
+
+        Returns:
+            True or False
+        """
+        try:
+            socket.inet_aton(ipv4)
+        except socket.error:
+            raise Exception("Invalid IPv4 address {}".format(ipv4))
+
+        try:
+            self.sonichost.shell("{}ping -q -c{} {} > /dev/null".format(
+                self._ns_arg, count, ipv4
+            ))
+        except RunAnsibleModuleFail:
+            return False
+        return True
+
+    def get_active_ip_interfaces(self):
+        """
+        Return a dict of active IP (Ethernet or PortChannel) interfaces, with
+        interface and peer IPv4 address.
+
+        Returns:
+            Dict of Interfaces and their IPv4 address
+        """
+        ip_ifs = self.show_ip_interface()["ansible_facts"]
+        ip_ifaces = {}
+        for k,v in ip_ifs["ip_interfaces"].items():
+            if (k.startswith("Ethernet") or
+                (k.startswith("PortChannel") and k.find("400") == -1)
+            ):
+                if (v["admin"] == "up" and v["oper_state"] == "up" and
+                        self.ping_v4(v["peer_ipv4"])
+                    ):
+                    ip_ifaces[k] = {
+                        "ipv4" : v["ipv4"],
+                        "peer_ipv4" : v["peer_ipv4"]
+                    }
+
+        return ip_ifaces
+
+    def bgp_drop_rule(self, ip_version, state="present"):
```

Comment:
At this point only QoS SAI test is using it, but I think BGP drop rule can be used by anyone and is not tied to QoS tests. Besides that the drop rule requires knowledge of Multi ASIC since it needs to be done under each ASIC namespace. It's better to avoid tests doing namespace specific commands, or having knowledge of multi ASIC, other than the ASIC fixture.